qiskit/algorithms/evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,192 @@
+                aux_ops_2,
+                state,
+            )
+            state = expm_multiply(A=-hamiltonian * timestep, B=state)
+            if np.nan in state:
+                raise ValueError(
+                    "An overflow has probably occured. Try increasing the amount of timesteps."
+                )
+            state = state / np.linalg.norm(state)
+
+        (ops_ev_mean[:, self.timesteps], ops_ev_std[:, self.timesteps],) = self._evaluate_aux_ops(
+            aux_ops,
+            aux_ops_2,
+            state,
+        )
+
+        aux_ops_history = self._create_observable_output(
+            ops_ev_mean,
+            ops_ev_std,
+            evolution_problem.aux_operators,
+        )
+        aux_ops = self._create_observable_output(
+            ops_ev_mean[:, self.timesteps],
+            ops_ev_std[:, self.timesteps],
+            evolution_problem.aux_operators,
+        )
+
+        return EvolutionResult(
+            evolved_state=StateFn(state), aux_ops_evaluated=aux_ops, observables=aux_ops_history
+        )
+
+    def _create_observable_output(
+        self,
+        ops_ev_mean: np.ndarray,
+        ops_ev_std: np.ndarray,
+        aux_ops: ListOrDict,
+    ) -> ListOrDict[Union[Tuple[np.ndarray, np.ndarray], Tuple[complex, complex]]]:
+        """Creates the right output format for the evaluated auxiliary operators."""
+        operator_number = 0 if aux_ops is None else len(aux_ops)
+        observable_evolution = [(ops_ev_mean[i], ops_ev_std[i]) for i in range(operator_number)]
+
+        if isinstance(aux_ops, dict):
+            observable_evolution = dict(zip(aux_ops.keys(), observable_evolution))
+
+        return observable_evolution
+
+    def _evaluate_aux_ops(
+        self,
+        aux_ops: List[sp.csr_matrix],
+        aux_ops_2: List[sp.csr_matrix],
+        state: np.ndarray,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Evaluates the aux operators if they are provided and stores their value.
+
+        Returns:
+            Tuple of the mean and standard deviation of the aux operators for a given state.
+        """
+        op_mean = np.array([np.real(state.conjugate().dot(op.dot(state))) for op in aux_ops])
+        op_std = np.sqrt(
+            np.array([np.real(state.conjugate().dot(op2.dot(state))) for op2 in aux_ops_2])
+            - op_mean**2
+        )
+        return op_mean, op_std
+
+    def _start(
+        self, evolution_problem: EvolutionProblem
+    ) -> Tuple[np.ndarray, sp.csr_matrix, sp.csr_matrix, List[sp.csr_matrix], float]:
+        """Returns a tuple with the initial state as an array and the operator needed for time
+            evolution as a sparse matrix.
+
+        Args:
+            evolution_problem: The definition of the evolution problem.
+
+        Returns:
+            A tuple with the initial state as an array, the operator needed for time
+            evolution as a sparse matrix and the number in which to divide the time evolution.
```

Comment:
This docstring doesn't seem to match the signature -- there's the aux_ops and aux_ops_2 missing I believe?